hooks/gen_docs.py

```diff
@@ -1,10 +1,263 @@
 """Generates the whole 'generatable' KPOps documentation"""
+import os
 import subprocess
+from pathlib import Path
+
+import yaml
 
 from hooks import PATH_ROOT
+from kpops.cli.registry import _find_classes
+from kpops.components.base_components.pipeline_component import PipelineComponent
+from kpops.utils.yaml_loading import load_yaml_file
 
 PATH_KPOPS_MAIN = PATH_ROOT / "kpops/cli/main.py"
 PATH_CLI_COMMANDS_DOC = PATH_ROOT / "docs/docs/user/references/cli-commands.md"
+PATH_DOCS_RESOURCES = PATH_ROOT / "docs/docs/resources"
+PATH_DOCS_COMPONENTS = PATH_DOCS_RESOURCES / "pipeline-components"
+PATH_DOCS_COMPONENTS_DEPENDENCIES = (
+    PATH_DOCS_COMPONENTS / "dependencies/pipeline_component_dependencies.yaml"
+)
+PATH_DOCS_COMPONENTS_DEPENDENCIES_DEFAULTS = (
+    PATH_DOCS_COMPONENTS / "dependencies/defaults_pipeline_component_dependencies.yaml"
+)
+PATH_DOCS_KPOPS_STRUCTURE = PATH_DOCS_COMPONENTS / "dependencies/kpops_structure.yaml"
+KPOPS_COMPONENTS = tuple(_find_classes("kpops.components", PipelineComponent))
+KPOPS_COMPONENTS_INHERITANCE_REF: dict = {
+    component.get_component_type(): component.__base__.get_component_type()  # type: ignore[reportGeneralTypeIssues]
```

Comment:
does this dict hold all kpops component classes with their direct parent/base class? what if it would inherit from multiple bases? (hypothetically, not sure if that's the case for any of the builtin components)

Reply:
we have something similar here, just FYI: https://github.com/bakdata/kpops/blob/2804fbec0e4fef86f0a0f65f4ec5843fd2fc69c8/kpops/components/base_components/base_defaults_component.py#L143
this could also be extracted into an independent function in case you'd wanna reuse it

Reply:
if you're going for the current approach you could help type checker out here and get rid of the pragma by using
```python
cast(type[PipelineComponent], component.__base__).get_component_type()
```

Reply:
Yes, it currently supports only single inheritance. If it were to encounter a class with multiple parents, it just [picks the first one](https://note.nkmk.me/en/python-issubclass-mro-bases-subclasses/).

The thing is,  `inspect.getmro()` returns `component.__mro__` which is a tuple of `component` + all parent classes up to `object` in method resolution order.
IMO if we wanna support multiple inheritance, easier would be `component.__bases__` which returns a tuple of all immediate parents.

I have to see how to solve the dependencies in such a scenario.

Reply:
I looked and while it is certainly possible and not hard to implement, I would prefer not to for now. I cannot find a way to avoid adding even more nested loops in `filter_sections()` or `filter_section()` and I doubt that we would need such a feature soon. I don't want to add complexity and worsen readability.

If the need arises, I'd be happy to add support for multiple inheritance, but I suspect that by then we would have moved to examples generation without the need for manually-maintained yamls. At that point this script would need to be heavily modified.